### PR TITLE
Migrate Lidar from STM32 to Raspberry Pi

### PR DESCRIPTION
**Pull request by ecourtois**

### CODE REVIEW

File: cogip/tools/copilot/sio_events.py [hunk 1]
```diff
@@ -80,3 +99,32 @@ async def on_wizard(self, sid, data):
...
+                pb_obstacle.radius = int(dyn_obstacle.radius)
+        await self._copilot.send_serial_message(server.obstacles_uuid, pb_obstacles)
+        await self.emit("obstacles", obstacles, room="dashboards")
+
+    async def on_lidar_data(self, sid, lidar_data):
+        """
+        Callback on lidar data.
+
+        In emulation mode, receive Lidar data from the Monitor,
+        and forward them the Detector in charge of computing dynamic obstacles.
```
Comment (gdoffe) [inline cogip/tools/copilot/sio_events.py:127]: `to the Detector`

File: cogip/tools/detector/detector.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    def stop_obstacles_updater(self) -> None:
+        """
+        Stop the thread updating obstacles list.
+        """
+        self._obstacles_updater_loop.stop()
+
+    def try_connect(self):
+        """
+        Poll to wait for the first copilot connection.
+        Deconnections/reconnections are handle directly by the client.
```
Comment (gdoffe) [inline cogip/tools/detector/detector.py:104]: Disconnections ?

File: docs/index.md [hunk 3]
```diff
@@ -7,21 +7,26 @@ This robot is developed for [Eurobot](https://www.eurobot.org/), French robotic
...
 
-`Copilot` received robot state and shell menus from `mcu-firmware` and emits SocketIO messages in JSON format to connected monitors.
+`Copilot` received robot state and shell menus from `mcu-firmware` and emits SocketIO messages in JSON format to connected `Monitor`/`Dashboards`.
 
-Monitors can send commands to `Copilot` which forwards them to `mcu-firmware`.
+`Monitor`/`Dashboards` can send commands to `Copilot` which forwards them to `mcu-firmware`.
 
 The touchscreen displays the `Copilot` dashboard using an web browser embedded in the Raspberry Pi.
 
+`Detector` reads Lidar data (for fake data sent by `Monitor` in emulation mode), builds dynamic obstacles
```
Comment (gdoffe) [inline docs/index.md:26]: s/for/or/

**Review by gdoffe — APPROVED** [review]
Awesome PR ! Few typos to fix but i already approve so you can merge once fixed.

File: cogip/utils/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from .threadloop import ThreadLoop  # noqa
```
[the remark this answers is not in the document]
Comment (ecourtois) [reply cogip/utils/__init__.py:1]: Yes, the import is here to expose the `ThreadLoop` class directly in the `cogip.utils` package.
The import not being used is this file, `# noqa` removes the warning about unused import.

File: cogip/tools/detector/detector.py (hunk 2 above)
Comment (ecourtois) [reply cogip/tools/detector/detector.py:104]: Done

File: docs/index.md (hunk 3 above)
Comment (ecourtois) [reply docs/index.md:26]: Done

**Pull request merged**